apps/extension/server/src/server.ts

```diff
@@ -195,17 +199,30 @@ async function handleDocumentChange(textDocument: TextDocument): Promise<void> {
       filterOutIgnoredMachines(extracted).machines,
     ]);
 
+    // Create typegen data for typed machines. This will throw if there are any errors.
     const types = machineResults
       .filter(
         (machineResult): machineResult is NonNullable<typeof machineResult> =>
-          !!machineResult?.machineCallResult.definition?.tsTypes?.node,
+          machineResult?.machineCallResult.definition?.tsTypes?.node !==
+          undefined,
       )
-      .map((machineResult, index) =>
-        getTypegenData(
+      .map((machineResult, index) => {
+        return getTypegenData(
           UriUtils.basename(URI.parse(textDocument.uri)),
           index,
           machineResult,
-        ),
+        );
+      });
+
+    // Iterate non-typed machines and throw if there are any errors.
+    machineResults
+      .filter(
+        (machineResult): machineResult is NonNullable<typeof machineResult> =>
+          machineResult?.machineCallResult.definition?.tsTypes?.node ===
+          undefined,
+      )
+      .forEach((machineResult) =>
+        introspectMachine(createIntrospectableMachine(machineResult) as any),
```

Comment:
```suggestion
        // for the time being we are piggy-backing on the fact that `createMachine` called in `instrospectMachine` might throw for invalid configs
        introspectMachine(createIntrospectableMachine(machineResult) as any),
```

Reply:
I wonder. Do we actually need to call `introspectMachine` here? Or is it enough to call `createInstrospectableMachine`? I'm not sure which one actually throws those errors.

Reply:
In my testing, I found that we do need to call `introspectMachine` before it started throwing on errors